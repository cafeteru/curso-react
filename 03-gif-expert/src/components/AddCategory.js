import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const defaultValue = "";
  const [inputValue, setInputValue] = useState(defaultValue);
  const handleInputChange = (e) => setInputValue(e.target.value);
  const handleSubmit = (event) => {
    event.preventDefault(); // Evita recarga de la pagina cuando envia el formulario
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue(defaultValue);
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = { setCategories: PropTypes.func.isRequired };
