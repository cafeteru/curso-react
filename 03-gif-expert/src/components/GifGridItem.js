export const GifGridItem = ({ title, url }) => {
  return (
    <div className={"card"}>
      <img alt={title} src={url} />
      <p>{title}</p>
    </div>
  );
};
