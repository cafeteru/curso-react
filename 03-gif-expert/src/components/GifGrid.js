import { useFetchGifs } from "../hooks/UseFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);
  return (
    <>
      <h3>{category}</h3>
      {loading && <p>"Cargando..."</p>}
      <div className={"card-grid"}>
        {gifs.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
