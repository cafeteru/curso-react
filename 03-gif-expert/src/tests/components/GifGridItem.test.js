import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("GifGridItem", () => {
  const title = "Titulo";
  const url = "http://localhost:8080/img.jpeg";
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el tÃ­tulo", () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(title);
  });

  test("Debe de tener una imagen con alt", () => {
    const img = wrapper.find("img");
    // prop -> para obtener los atributos de la etiqueta
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe de tener la clase card", () => {
    const div = wrapper.find("div");
    // props -> devuelve un objeto con todos los atributos de la etiqueta
    expect(div.props().className).toContain("card");
  });
});
