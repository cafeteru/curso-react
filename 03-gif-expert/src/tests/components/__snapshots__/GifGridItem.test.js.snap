// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="http://localhost:8080/img.jpeg"
  />
  <p>
    Titulo
  </p>
</div>
`;
